### PR TITLE
validation working

### PR DESCRIPTION
**Pull request by rwobben**
Solves #24

### CODE REVIEW

File: paytime/src/Lib.hs [hunk 1]
```diff
@@ -47,14 +46,24 @@ doCreateAccounts accts = do
...
+  case readMaybe balStr of
+     Just bal -> return bal
+     Nothing -> do 
+                  putStrLn "That wasn't a number. Please try again."
+                  getBalance
+
+
+
 addAccount :: Account -> AccountStore -> AccountStore
 addAccount newAcct = Map.insert (accountName newAcct) newAcct
```
Comment (GetContented) [inline paytime/src/Lib.hs]: this can be eta-reduced. That is, you can remove newAcct from either side of the = sign.

Comment (rwobben) [inline paytime/src/Lib.hs]: I do not think so.

When I change it to : 

addAccount :: Account -> AccountStore -> AccountStore
addAccount = Map.insert (accountName)

I see this error message : 

    /home/ubuntu/workspace/paytime/paytime/src/Lib.hs:68:26:
        Couldn't match type ‘Account -> String’ with ‘[Char]’
        Expected type: String
          Actual type: Account -> String
        Probable cause: ‘accountName’ is applied to too few arguments
        In the first argument of ‘Map.insert’, namely ‘(accountName)’

When I change it to : 

addAccount :: Account -> AccountStore -> AccountStore
addAccount = Map.insert (accountName newAcct)

Then I see a error message that newAcct is not known

Comment (GetContented) [inline paytime/src/Lib.hs]: Sorry, you're right of course. I didn't look properly enough! :) the application of accountName is using newAcct, so obviously you can't eta-reduce it. Sorry :)

Comment (rwobben) [inline paytime/src/Lib.hs]: Np, So we learn both . 
Commit it

Comment (GetContented) [inline paytime/src/Lib.hs]: Haha, not quite :) I just didn't read it properly (you're usually awake when I'm very very tired and just about to go to sleep)

**Pull request merged**